packages/get-starknet/index.ts

```diff
@@ -0,0 +1,80 @@
+import {  MetaMaskSnap } from './src/snap';
+import {  MetaMaskSnapWallet } from './src/wallet';
+
+const init = async() => {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const provider =  await MetaMaskSnap.GetProvider(window as any) as any;
+    const wallet = new MetaMaskSnapWallet(provider, '*');
+    
+    const address = await wallet.enable();
+    console.log("address", address)
+
+    const wallet_switchStarknetChain = await wallet.request( {
+        type: 'wallet_switchStarknetChain',
+        params: {
+            chainId: '0x534e5f4d41494e',
+        },
+    });
+    console.log("wallet_switchStarknetChain", wallet_switchStarknetChain)
+
+    const wallet_addStarknetChain = await wallet.request( {
+        type: 'wallet_addStarknetChain',
+        params: {
+            chainId: '0x534e5f4d41494e',
+            chainName: 'StarkNet',
+            rpcUrls: ['https://stg-rpc.starknet.io'],
+            blockExplorerUrls: ['https://stg-explorer.starknet.io'],
+        },
+    });
+    console.log("wallet_addStarknetChain", wallet_addStarknetChain)
+
+    const message = {
+        types: {
+            StarkNetDomain: [
+                { name: "name", type: "string" },
+                { name: "version", type: "felt" },
+                { name: "chainId", type: "felt" },
+            ],
+            Airdrop: [
+                { name: "address", type: "felt" },
+                { name: "amount", type: "felt" }
+            ],
+            Validate: [
+                { name: "id", type: "felt" },
+                { name: "from", type: "felt" },
+                { name: "amount", type: "felt" },
+                { name: "nameGamer", type: "string" },
+                { name: "endDate", type: "felt" },
+                { name: "itemsAuthorized", type: "felt*" }, // array of felt
+                { name: "chkFunction", type: "selector" }, // name of function
+                { name: "rootList", type: "merkletree", contains: "Airdrop" } // root of a merkle tree
+            ]
+        },
+        primaryType: "Validate",
+        domain: {
+            name: "myDapp", 
+            version: "1",
+            chainId: "0x534e5f474f45524c49", 
+        },
+        message: {
+            id: "0x0000004f000f",
+            from: "0x2c94f628d125cd0e86eaefea735ba24c262b9a441728f63e5776661829a4066",
+            amount: "400",
+            nameGamer: "Hector26",
```

Comment:
Is this Test data ?
Can you add some inline code comments as to what is happening in this file ?